### PR TITLE
Wall

### PR DESCRIPTION
**Pull request by JensWehner**

### CODE REVIEW

File: src/libxtp/gwbse/gwbseengine.cc [hunk 1]
```diff
@@ -148,9 +148,9 @@ namespace votca {
             // parse DFT data, if required
             if (_do_dft_parse) {
                 if (_redirect_logger) {
-                    CTP_LOG(ctp::logINFO, _gwbse_engine_logger) << "Parsing DFT data from " << _dftlog_file << " and " << _MO_file << flush;
+                    CTP_LOG_SAVE(ctp::logINFO, _gwbse_engine_logger) << "Parsing DFT data from " << _dftlog_file << " and " << _MO_file << flush;
```
Comment (JensWehner) [inline src/libxtp/gwbse/gwbseengine.cc]: I do not understand why this is necessary?

Comment (baumeier) [inline src/libxtp/gwbse/gwbseengine.cc]: For what it is worth, neither do I...

Comment (junghans) [inline src/libxtp/gwbse/gwbseengine.cc]: `CTP_LOG` is a macro defined in `include/votca/ctp/logger.h`:
```c
#define CTP_LOG(level, log) \
if ( &log != NULL && level > (log).getReportLevel() ) ; \
else (log)(level)
```
and in this function `log` is never `NULL` and hence the compiler complains about a useless comparison:
 ```
/home/votca/votca/xtp/src/libxtp/gwbse/gwbseengine.cc: In member function 'void votca::xtp::GWBSEENGINE::ExcitationEnergies(votca::xtp::QMPackage*, std::vector<votca::ctp::Segment*>, votca::xtp::Orbitals*)':
/home/votca/votca/ctp/include/votca/ctp/logger.h:35:11: warning: the address of '_gwbse_engine_logger' will never be NULL [-Waddress]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/home/votca/votca/ctp/include/votca/ctp/logger.h:35:11: warning: the address of '_gwbse_engine_logger' will never be NULL [-Waddress]
```
(https://travis-ci.org/votca/votca/jobs/260781933#L5116)

Comment (junghans) [inline src/libxtp/gwbse/gwbseengine.cc]: On a related note, we should switch to `boost::logging`.

**Comment by junghans** [comment]
Now all warnings are gone in votca/votca#19, so ready to merge.

**Pull request merged**